infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -67,49 +77,161 @@ public class LogReplicationConfigManager {
+     * Used for non-upgrade testing purpose only. Note that this constructor will keep the version table in
+     * uninitialized state, in which case LR will be constantly considered to be not upgraded.
      */
     @VisibleForTesting
     public LogReplicationConfigManager(CorfuRuntime runtime) {
+        this.rt = runtime;
         this.corfuStore = new CorfuStore(runtime);
         this.pluginConfigFilePath = EMPTY_STR;
     }
 
     public LogReplicationConfigManager(CorfuRuntime runtime, String pluginConfigFilePath) {
+        this.rt = runtime;
         this.pluginConfigFilePath = pluginConfigFilePath;
         this.corfuStore = new CorfuStore(runtime);
+        initLogReplicationVersionPlugin();
+        setupVersionTable();
+    }
+
+    private void initLogReplicationVersionPlugin() {
+        log.info("Plugin :: {}", pluginConfigFilePath);
+        LogReplicationPluginConfig config = new LogReplicationPluginConfig(pluginConfigFilePath);
+        File jar = new File(config.getStreamFetcherPluginJARPath());
+        try (URLClassLoader child = new URLClassLoader(new URL[]{jar.toURI().toURL()}, this.getClass().getClassLoader())) {
+            Class plugin = Class.forName(config.getStreamFetcherClassCanonicalName(), true, child);
+            logReplicationVersionAdapter = (ILogReplicationVersionAdapter) plugin.getDeclaredConstructor()
+                    .newInstance();
+        } catch (Exception e) {
+            log.error("Fatal error: Failed to get Stream Fetcher Plugin", e);
+            throw new UnrecoverableCorfuError(e);
+        }
+    }
+
+    /**
+     * Read registry table's entries into memory for config synchronization.
+     */
+    public void refreshInMemoryRegistryTableEntries() {
+        int numRetries = 9;
+        while (true) {
+            try {
+                CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>> registryTable =
+                        rt.getTableRegistry().getRegistryTable();
+                entryList = registryTable.entryStream().collect(Collectors.toList());
+                break;
+            } catch (Exception e) {
+                if (numRetries-- <= 0) {
+                    throw e;
+                }
+                log.info("Exception caught when syncing with registry table, will retry {} times.", numRetries);
+            }
+        }
+    }
+
+
+    /**
+     * Get streams to replicate.
+     *
+     * @return Set of fully qualified names of streams to replicate
+     */
+    public Set<String> getStreamsToReplicate() {
+        Set<String> streamNameSet = new HashSet<>();
+        // Get streams to replicate from registry table and add to streamNameSet.
+        entryList.stream()
+                .filter(entry -> entry.getValue().getMetadata().getTableOptions().getIsFederated())
+                .map(entry -> getFullyQualifiedTableName(entry.getKey()))
+                .forEachOrdered(streamNameSet::add);
+        // Add registryTable to the streams
```

Comment:
nit - can you use MERGE_ONLY_STREAMS instead of hardcoding the 2 tables here?

Reply:
I think for now we have to manually add them here, the MERGE_ONLY_STREAMS are UUIDs